ipv-core/src/main/java/com/ibm/whc/deid/providers/masking/BinningMaskingProvider.java

```diff
@@ -55,13 +68,27 @@ public String mask(String identifier) {
       debugFaultyInput("identifier");
       return null;
     }
-
+    
     double value;
     try {
       value = Double.parseDouble(identifier);
     } catch (NumberFormatException e) {
       // For this provider, we do not return a random value
       return applyUnexpectedValueHandling(identifier, null);
+    }  
+
+    if (this.useSingleBucketOverThreshold==true && this.useSingleBucketUnderThreshold==true) {
+    	
+        double upperRange=this.singleBucketOverThresholdValue;
+        double lowerRange=this.singleBucketUnderThresholdValue;
+        
+        if(value>=upperRange) {
```

Comment:
Note that you don't need to create the local variables upperRange and lowerRange.  You can just refer to this.singleBucketOverThresholdValue and this.singleBucketUnderThresholdValue directly in the "if" statements.  This isn't wrong.  It is a personal style thing.  I normally prefer to keep the number of variables reduced where possible as it is less code to read through.  In this simple method, it doesn't really matter.  If this was very complicated, though, it might help.